homeassistant/components/ping/config_flow.py

```diff
@@ -0,0 +1,57 @@
+"""Config flow for Ping (ICMP) integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import Any, cast
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.helpers import selector
+from homeassistant.helpers.schema_config_entry_flow import (
+    SchemaConfigFlowHandler,
+    SchemaFlowFormStep,
+)
+
+from .const import CONF_PING_COUNT, DEFAULT_PING_COUNT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(
+            CONF_PING_COUNT, default=DEFAULT_PING_COUNT
+        ): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                min=1, max=100, mode=selector.NumberSelectorMode.BOX
+            )
+        ),
+    }
+)
+
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): selector.TextSelector(),
+    }
```

Comment:
Generally we try to avoid collecting a name if we can.

Another option is to use the config entry title to generate the name so when they rename the config entry, it would generate a reload ie `entry.async_on_unload(entry.add_update_listener(_async_update_listener))`

We use this pattern in a few other integrations to still give the user the flexibility to rename and its also nice not to have to manage the name in two places

Reply:
There is an example of this in `flux_led`.  I think `wled` might do this as well

Reply:
Thanks a lot, will have a look